### ISSUE TITLE
Can only send email to addresses in the Thunderbird address book

### ISSUE THREAD

**Issue by johnlane** [absorb-it/Virtual-Identity#16]
Recently I can only send email to addresses in the Thunderbird address book. Any other addresses result in a pop-up dialog containing:

    virtualIdentity Extension Error
    sorry for the inconveniance
    try to save your email and restart!
    please send the bug-report to fix this issue

I have Virtual Identity 0.9.36 last updated 25.nov.2017 ant Thunderbird 52.4.0 (64-bit) on Arch Linux.

**Comment by absorb-it** [comment]
Hi, works for me with the same environment...
There is no chance to say anything without a bug report, after this message there should be some open mail-window with the relevant information targeted to irtualIdentityBug at absorb.it? Please send this report.

Regards, Rene

**Comment by johnlane** [comment]
Sorry, I had meant to post that then forgot. 

Steps used to reproduce

1. press *Write* to open a new mail dialog
2. enter `bob@example.com` in the `To` box
3. enter `test` in the subject field
4. enter `test` in the body
5. press `send`

I can not send the bug report email. I get the same error dialog. Here is the bug report message body:
    
    virtual identity v 0.9.36:
    
    # please send this debug-information if possible #
    # it will help to make the extension better #
    ----------------------------------------------
    (even if some other message compose window does
    not work anymore, sending this message might even
    work with a virtual identity)
    ----------------------------------------------
    
    
    virtualIdentity raised an error: [Exception... "Component returned failure code: 0x80004001 (NS_ERROR_NOT_IMPLEMENTED) [nsIMsgCompose.expandMailingLists]" nsresult: "0x80004001 (NS_ERROR_NOT_IMPLEMENTED)" location: "JS frame :: chrome://messenger/content/messengercompose/MsgComposeCommands.js :: expandRecipients :: line 4466" data: no]
    
    expandRecipients@chrome://messenger/content/messengercompose/MsgComposeCommands.js:4466:3
    GenericSendMessage@chrome://messenger/content/messengercompose/MsgComposeCommands.js:2940:5
    main.replacement_functions.GenericSendMessage@chrome://v_identity/content/v_identity.js:175:15
    SendMessage@chrome://messenger/content/messengercompose/MsgComposeCommands.js:3280:3
    defaultController.commands.cmd_sendButton.doCommand@chrome://messenger/content/messengercompose/MsgComposeCommands.js:703:11
    defaultController.doCommand@chrome://messenger/content/messengercompose/MsgComposeCommands.js:875:5
    goDoCommand@chrome://global/content/globalOverlay.js:100:7
    oncommand@chrome://messenger/content/messengercompose/messengercompose.xul:1:1
    
    
    
    error-log:
    ---------

**Comment by absorb-it** [comment]
Hi,

On 28.11.2017 15:25, johnlane wrote:
> error: [Exception... "Component returned failure code: 0x80004001
> (NS_ERROR_NOT_IMPLEMENTED) [nsIMsgCompose.expandMailingLists]" nsresult:
> "0x80004001 (NS_ERROR_NOT_IMPLEMENTED)" location: "JS frame ::
> chrome://messenger/content/messengercompose/MsgComposeCommands.js ::
> expandRecipients :: line 4466" data: no]

this is strange, the error is raised from thunderbird core and seems not
related to the extension. I assume that you can send mail if the
extension is disabled?

If thats the case than please enable complete debugging, something big
must go wrong before the mail should be send. Please enable "Debug Log"
in the last Tab of the Preferences, best save it to your /tmp directory.
Close and restart Thunderbird and send mail with virtual identity. Than
disable the log and check this file for errors. Before posting (or
better sending to me as attachment), check for confidential information
in this log!!!

Sorry, this is really troublesome, but I have no other idea. Best regards,
Rene

**Comment by johnlane** [comment]
I'm going to look into this a bit more. I just discovered that I can not send mail to an address that is not in the address book even with the extension disabled. But I get no error at all - the send button doesn't do anything.... I'll update when I know more and probably close this out....

**Comment by absorb-it** [comment]
Hi,

On 28.11.2017 16:00, johnlane wrote:
> I'm going to look into this a bit more. I can not send mail to an
> address that is not in the address book even with the extension
> disabled. But I get no error at all - the send button doesn't do
> anything....

right, than this is not related to this extension. Its only that the
extension will report this error (instead of being quit). You will see
the same error in the Error Console (check Tools Menu - Developer Tools)
with disabled extension after you pressed the send button (and nothing
happened)

Best regards, Rene

**Comment by johnlane** [comment]
Thanks, that pointed me to the right place to see the error messages (the Error Console) which I was struggling to find. That's really helpful. I'll go off and see if I can discover what the problem is...

**Comment by johnlane** [comment]
I've resolved it. The Virtual Identity extension is fine.

For background there is an obscure [Mozilla Bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1371309) that breaks mail sending when an LDAP directory is defined at `Preferences > Composotion > Addressing` even if directory server use is disabled.

Luckily for me the directory I had defined was a left-over from the past and I didn't need it. After deleting it everything worked fine, as expected.

So happy days :)
